core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultTopologyMonitor.java

```diff
@@ -215,9 +263,17 @@ private NodeInfo asNodeInfo(AdminRow row) {
     if (broadcastAddress == null) {
       broadcastAddress = row.getInetAddress("peer"); // in system.peers
     }
-    builder.withBroadcastAddress(broadcastAddress);
-
-    builder.withListenAddress(row.getInetAddress("listen_address"));
+    int broadcastPort = 0;
+    if (row.contains("peer_port")) {
+      broadcastPort = row.getInteger("peer_port");
+    }
```

Comment:
:+1:.  The 3.x PR creates a new type [`InetAddressOptPort`](https://github.com/aweisberg/java-driver/blob/648fb67fd14ef0b263b90d50a912c724576a924a/driver-core/src/main/java/com/datastax/driver/core/utils/InetAddressOptPort.java).  Greg and I talked about it and agreed it would be better to just assign the port to 0 in the case it can't be discerned instead of creating a new type.  I agree that we should document these methods and indicate the port can be 0 if it can't be determiend.